pubnub-gson/src/main/java/com/pubnub/internal/v2/subscription/EmitterHelper.kt

```diff
@@ -0,0 +1,90 @@
+import com.pubnub.api.models.consumer.objects_api.membership.PNMembershipResult
+import com.pubnub.api.models.consumer.objects_api.uuid.PNUUIDMetadataResult
+import com.pubnub.api.models.consumer.pubsub.PNMessageResult
+import com.pubnub.api.models.consumer.pubsub.PNPresenceEventResult
+import com.pubnub.api.models.consumer.pubsub.PNSignalResult
+import com.pubnub.api.models.consumer.pubsub.files.PNFileEventResult
+import com.pubnub.api.models.consumer.pubsub.message_actions.PNMessageActionResult
+import com.pubnub.api.v2.callbacks.BaseEventEmitter
+import com.pubnub.api.v2.callbacks.EventListener
+import com.pubnub.internal.v2.callbacks.DelegatingEventListener
+import com.pubnub.internal.v2.callbacks.EventListenerCore
+
+class EmitterHelper(eventEmitter: BaseEventEmitter<EventListenerCore>) {
+    init {
+        eventEmitter.addListener(
+            DelegatingEventListener(
+                object : EventListener {
+                    override fun message(
+                        pubnub: PubNub,
+                        result: PNMessageResult,
+                    ) {
+                        onMessage?.invoke(result)
+                    }
+
+                    override fun presence(
+                        pubnub: PubNub,
+                        result: PNPresenceEventResult,
+                    ) {
+                        onPresence?.invoke(result)
+                    }
+
+                    override fun signal(
+                        pubnub: PubNub,
+                        result: PNSignalResult,
+                    ) {
+                        onSignal?.invoke(result)
+                    }
+
+                    override fun messageAction(
+                        pubnub: PubNub,
+                        result: PNMessageActionResult,
+                    ) {
+                        onMessageAction?.invoke(result)
+                    }
+
+                    override fun uuid(
+                        pubnub: PubNub,
+                        result: PNUUIDMetadataResult,
+                    ) {
+                        onUuid?.invoke(result)
+                    }
+
+                    override fun channel(
+                        pubnub: PubNub,
+                        result: PNChannelMetadataResult,
+                    ) {
+                        onChannel?.invoke(result)
+                    }
+
+                    override fun membership(
+                        pubnub: PubNub,
+                        result: PNMembershipResult,
+                    ) {
+                        onMembership?.invoke(result)
+                    }
+
+                    override fun file(
+                        pubnub: PubNub,
+                        result: PNFileEventResult,
+                    ) {
+                        onFile?.invoke(result)
+                    }
+                },
+            ),
+        )
+    }
+
+    var onMessage: ((PNMessageResult) -> Unit)? = null
```

Comment:
these could use the types you created, for example `var onMessage: OnMessageHandler? = null`
no need to use Kotlin functional types that return Unit
that would also let you simplify code that sets these handlers in the other classes (see other comment)

Reply:
Good point. I will change it.